packages/vite-plugin-svelte/src/utils/dependencies.ts

```diff
@@ -187,12 +187,23 @@ export function needsOptimization(dep: string, localRequire: NodeRequire): boole
 	const pkg = depData.pkg;
 	// only optimize if is cjs, using the below as heuristic
 	// see https://github.com/sveltejs/vite-plugin-svelte/issues/162
-	const isCjs = pkg.main && !pkg.module && !pkg.exports;
-	if (!isCjs) return false;
-	// ensure entry is js so vite can prebundle it
-	// see https://github.com/sveltejs/vite-plugin-svelte/issues/233
-	const entryExt = path.extname(pkg.main);
-	return !entryExt || entryExt === '.js' || entryExt === '.cjs';
+	if (pkg.type !== 'module' && !pkg.module) {
+		let entry;
+		// if exports is defined it takes precedence over main!
+		if (pkg.exports) {
+			// exports map syntax is too complex to try and find the correct one, let node do it's thing
+			entry = localRequire.resolve(dep);
```

Comment:
I think for `pkg.exports` we can safely assume that most packages export an ESM variant. `exports` for CJS-only sounds rare and probably not worth the effort to check for these.

I think the previous heuristic we have is fine, with an extra change on the final extension check like:

```js
return !entryExt || (entryExt === '.js' && pkg.type !== 'module') || entryExt === '.cjs';
```

Reply:
Another risk with using resolve might be that it's resolved based on node conditions, but during prebundling, it'll be bundled for the browser. This won't likely be a problem, but just something I thought of.

Reply:
that is true. Wonder if we can set the browser condition

Reply:
unfortunately it is not possible for us to set custom user conditions apart from passing a flag on the cli :see_no_evil:  https://nodejs.org/api/packages.html#resolving-user-conditions